src/corner-kick/src/service/RosConnector.ts

```diff
@@ -0,0 +1,132 @@
+        // We initialize a new array if this is the first callback for a given topic
+        if(this.topicsSubscriptions[topic] === undefined) {
+            this.topicsSubscriptions[topic] = [callback];
+        } else {
+            this.topicsSubscriptions[topic].push(callback);
+        }
+    }
+
+    /**
+     * Unsubscribe from a given topic.
+     * @param topic the topic to unsubscribe from
+     * @param callback the callback function to remove
+     */
+    public unsubscribeToTopic = (topic: string, callback: (topic: any) => void) => {
+        if(this.topicsSubscriptions[topic] !== undefined) {
+            const index = this.topicsSubscriptions[topic].indexOf(callback);
+            this.topicsSubscriptions[topic].splice(index, 1);
+        }
+    }
+
+    public publish = (name: string, messageType: string, message: any) => {
+        const topic = new ROSLib.Topic({
+            messageType,
+            name,
+            ros: this.ros,
+        });
+
+        topic.publish(message);
+    }
+
+    /**
+     * Internal method to subscribe to all ROS topics. This is the method that actually "subscribes" in a ROS sense. We do this to create an abstraction
+     * layer between ROS and the rest of our application
+     */
+    private subscribeToROS = () => {
+        // Request ROS to give us a list of all topics.
+        this.ros.getTopics(this.onTopicsReceived);
+
+        // Every 10 seconds, ask for a new list from ROS. If new topics are added during runtime, we will pick them up.
+        setTimeout(this.subscribeToROS, 10000);
+    }
+
+    /**
+     * This internal method subscribes (in a ROS sense) to every topic found. When a new message is received, it calls
+     * onMessageReceived, which does the actual act of notifying the rest of the application.
+     */
+    private onTopicsReceived = (response: {topics: string[], types: string[]}) => {
+        const newTopics: string[] = [];
+
+        response.topics.forEach((element, index) => {
+            // We only subscribe to topics we have not seen yet
+            if(this.topics.indexOf(element) === -1) {
+                const topic = new ROSLib.Topic({
+                    messageType: response.types[index],
+                    name: element,
+                    ros: this.ros,
+                });
+                topic.subscribe(this.onMessageReceived(element));
+            }
+            // We add the topic to the list of topics seen
+            newTopics.push(element);
+        });
+
+        this.topics = newTopics;
+    }
+
+    /**
+     * Method called when a new message is received from a ROS topic. Does the actual work of calling the user-provided
+     * callback functions.
+     */
+    private onMessageReceived = (name: string) => (message: any) => {
+        if(this.topicsSubscriptions[name] !== undefined) {
+            this.topicsSubscriptions[name].forEach((callback) => callback(message));
+        }
```

Comment:
We should maybe log some sort of error here if we don't have a subscription for this topic?